docs/src/tutorials/transitioning/transitioning_from_matlab.jl

```diff
@@ -270,13 +270,58 @@ objective_value(model)
 # end
 # ```
 
+# ## Symmetric and Hermitian matrices
+
+# Julia has specialized support for symmetric and Hermitian matrices in the
+# `LinearAlgebra` package:
+
+import LinearAlgebra
+
+# If you have a matrix that is numerically symmetric:
+
+x = [1 2; 2 3]
+
+#-
+
+LinearAlgebra.issymmetric(x)
+
+# then you can wrap it in a `LinearAlgebra.Symmetric` matrix to tell Julia's
+# type system that the matrix is symmetric.
+
+LinearAlgebra.Symmetric(x)
+
+# Using a `Symmetric` matrix lets Julia and JuMP use more efficient algorithms
+# when they are working with symmetric matrices.
+
+# If you have a matrix that is nearly but not exactly symmetric:
+
+x = [1.0 2.0; 2.001 3.0]
+LinearAlgebra.issymmetric(x)
+
+# then you could, as you might do in MATLAB, make it numerically symmetric as
+# follows:
+
+x_sym = 0.5 * (x + x')
+
+# In Julia, you can explicitly choose whether to use the lower or upper triangle
+# of the matrix:
+
+x_sym = LinearAlgebra.Symmetric(x, :L)
+
+#-
+
+x_sym = LinearAlgebra.Symmetric(x, :U)
+
+# The same applies for Hermitian matrices, using `LinearAlgebra.Hermitian` and
+# `LinearAlgebra.ishermitian`.
+
```

Comment:
What I wanted to point out here is that with MATLAB you have to be paranoid about making your matrices exactly Hermitian, in order to not get subtle errors, whereas with JuMP you just need to wrap them and you're good to go.

But now I've tested it, and it turns out that it's not true. If you try to constrain a non-Hermitian matrix to be in HermitianPSDCone() you get an error message, as I expected, but if you try to constrain a non-Symmetric matrix to be in PSDCone() JuMP will let you (and fail afterwards, of course). Is that intentional?

Reply:
Why? Is there any case where you would need that? As far as I can see this can either generate less efficient code or give you an error.

Reply:
I was just curious, I wasn't suggesting to change this, it would clearly be a breaking change.

Although now that you brought it up, I will definitely suggest removing it when the time comes for JuMP 2.0.

Reply:
> I will definitely suggest removing it when the time comes for JuMP 2.0

My goal is not to release JuMP 2.0 if at all possible. If we did, it would be for a minimal set of breaking changes, not just to remove some things we don't like. The only candidate was the nonlinear changes, but we seem to have managed to sneak them in with minimal disruption. The JuMP developers have no plans for JuMP 2.0.